### PR TITLE
Cgmes conversion regulating control uniformization

### PR DESCRIPTION
**Pull request by marqueslanauja**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Refactoring

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

Review all regulatingControl code and apply all reviewer comments. Code uniformization.
`HasCorrectlySet`: use a `boolean` instead of  `Map<String, Boolean> idsEq = new HashMap<>()`

### CODE REVIEW

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/RegulatingControlMappingForGenerators.java [hunk 1]
```diff
@@ -34,27 +34,33 @@ public RegulatingControlMappingForGenerators(RegulatingControlMapping parent) {
...
 
-    public void add(String iidmId, PropertyBag sm) {
-        String rcId = parent.getRegulatingControlId(sm);
+    public void add(String generatorId, PropertyBag sm) {
+        String cgmesRegulatingControlId = parent.getRegulatingControlId(sm);
         double qPercent = sm.asDouble(QPERCENT);
-        add(iidmId, rcId, qPercent);
+
+        if (mapping.containsKey(generatorId)) {
+            throw new CgmesModelException("Generator already added, IIDM Generator Id : " + generatorId);
```
Comment (MioRtia) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/RegulatingControlMappingForGenerators.java]: Typo: no space for `:`

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/RegulatingControlMappingForStaticVarCompensators.java [hunk 2]
```diff
@@ -123,10 +123,10 @@ private void setRegulatingControl(RegulatingControl control, StaticVarCompensato
         svc.setReactivePowerSetPoint(targetReactivePower);
         svc.setRegulationMode(regulationMode);
 
-        control.hasCorrectlySetEq(svc.getId());
+        control.hasCorrectlySet();
```
Comment (MioRtia) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/RegulatingControlMappingForStaticVarCompensators.java]: Same remark as above. This method should only indicates that it has been correctly set **for this equipment**.

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/RegulatingControlMappingForTransformers.java [hunk 3]
```diff
@@ -180,7 +180,7 @@ private void setRatioTapChangerControl(boolean regulating, CgmesRegulatingContro
                     "Unsupported regulation mode for Ratio tap changer. Considered as a fixed ratio tap changer.");
         }
 
-        control.hasCorrectlySetEq(rc.id);
+        control.hasCorrectlySet();
```
Comment (MioRtia) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/RegulatingControlMappingForTransformers.java]: Same remark.

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/RegulatingControlMappingForTransformers.java [hunk 4]
```diff
@@ -217,7 +217,7 @@ private void setPhaseTapChangerControl(CgmesRegulatingControlPhase rc, Regulatin
             context.fixed(control.mode, "Unsupported regulating mode for Phase tap changer. Considered as FIXED_TAP");
         }
 
-        control.hasCorrectlySetEq(rc.id);
+        control.hasCorrectlySet();
```
Comment (MioRtia) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/RegulatingControlMappingForTransformers.java]: Same remark.

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/RegulatingControlMapping.java [hunk 5]
```diff
@@ -70,8 +69,8 @@ public Context context() {
             this.targetDeadband = p.asDouble("targetDeadband", Double.NaN);
         }
 
-        void hasCorrectlySetEq(String id) {
-            idsEq.put(id, true);
+        void hasCorrectlySet() {
+            correctlySet = true;
```
[the remark this answers is not in the document]
Comment (marqueslanauja) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/RegulatingControlMapping.java]: We lost this behaviour in the generatorRegulatingControl. Refactored according with your comments.

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/RegulatingControlMappingForStaticVarCompensators.java (hunk 2 above)
Comment (marqueslanauja) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/RegulatingControlMappingForStaticVarCompensators.java]: Refactored according with your comments.

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/RegulatingControlMappingForTransformers.java (hunk 4 above)
Comment (marqueslanauja) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/RegulatingControlMappingForTransformers.java]: Refactored according with your comments.

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/RegulatingControlMapping.java [hunk 6]
```diff
@@ -91,15 +90,10 @@ public void setAllRegulatingControls(Network network) {
...
-            }
-            for (Map.Entry<String, Boolean> e : entry.getValue().idsEq.entrySet()) {
-                if (!e.getValue()) {
-                    return false;
-                }
+            if (entry.getValue().correctlySet) {
+                return true;
             }
-            return true;
+            return false;
```
[the remark this answers is not in the document]
Comment (marqueslanauja) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/RegulatingControlMapping.java]: Refactored according with your comments.

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/RegulatingControlMappingForGenerators.java (hunk 1 above)
Comment (marqueslanauja) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/RegulatingControlMappingForGenerators.java]: Done

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/RegulatingControlMappingForTransformers.java (hunk 3 above)
Comment (marqueslanauja) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/RegulatingControlMappingForTransformers.java]: Refactored according with your comments.

**Pull request merged**